root_pandas/readwrite.py

```diff
@@ -147,6 +148,18 @@ def do_flatten(arr):
         arr = append_fields(arr_, '__array_index', idx, usemask=False, asrecarray=True)
         return arr
 
+    def remove_high_dimensions(arr):
+        allowed_dimensions = [0]
+        first_row = arr[0]
+        good_cols = [True if x.ndim in allowed_dimensions else False for x in first_row]
+        col_names = np.array(list(arr.dtype.names))
+        good_names = col_names[np.array(good_cols)]
+        bad_names = col_names[np.array([not x for x in good_cols])]
+        for bad_name in bad_names:
+            warnings.warn("Dropped {bad_name} branch because dimension is unfit for DataFrame"
+                          .format(bad_name=bad_name), UserWarning)
+        return arr[good_names]
```

Comment:
This might be a bit of a problem.
We definitely want to avoid making a copy here.
At least on my system, this creates a copy and emits a `FutureWarning` that this might change in the future.